src/website/shared/management/commands/ingest_delta_cve.py

```diff
@@ -19,11 +20,72 @@
 logger = logging.getLogger(__name__)
 
 
+def ingest_day(repo: Repository, day: datetime.datetime) -> CveIngestion:
+    # Fetch the latest daily release
+    try:
+        release = repo.get_release(f"cve_{day}_at_end_of_day")
+    except UnknownObjectException:
+        raise CommandError(f"No end of day release for {day} found.")
+
+    logger.info(f"Fetched release: {release.title}")
+
+    # Get the bulk cve list asset
+    bundle = release.assets[0]
+
+    if bundle.name != f"{day}_delta_CVEs_at_end_of_day.zip":
+        logger.error(f"Wrong delta asset: {bundle.name}")
+
+        raise CommandError("Unable to get delta CVEs.")
+
+    # Create a temporary directory to work in
+    with tempfile.TemporaryDirectory() as tmp_dir:
+        tmp_arc = f"{tmp_dir}/deltaCves.zip"
+
+        # Download the zip file
+        logger.info(f"Downloading the delta: {bundle.name}")
+        r = requests.get(bundle.browser_download_url)
+
+        if r.status_code != 200:
+            raise CommandError(f"Unable to download the delta, error {r.status_code}")
+
+        with open(tmp_arc, "wb") as fz:
+            fz.write(r.content)
+
+        # Extract the archive
+        with zipfile.ZipFile(tmp_arc) as z_arc:
+            logger.info("Extracting the first archive to cves.zip")
+
+            z_arc.extractall(path=tmp_dir)
+
+        # Open a single transaction for the db
+        with transaction.atomic():
+            # Traverse the tree and import cves
+            cve_list = glob(f"{tmp_dir}/deltaCves/*.json")
+            logger.info(f"{len(cve_list)} CVEs to ingest.")
+
+            for j_cve in cve_list:
+                with open(j_cve) as fc:
+                    data = json.load(fc)
+                    cve_id = data["cveMetadata"]["cveId"]
+
+                    make_cve(
+                        data,
+                        record=models.CveRecord.objects.filter(cve_id=cve_id).first(),
+                    )
+
+    # Record the ingestion
+    logger.info(f"Saving the ingestion valid up to {day}")
+
+    return CveIngestion.objects.create(valid_to=day, delta=True)
+
+
 class Command(BaseCommand):
     help = "Ingest CVEs in bulk using the Mitre CVE repo"
 
     def add_arguments(self, parser: argparse.ArgumentParser) -> None:
-        parser.add_argument("date", help="Date of the delta to download.")
+        parser.add_argument(
+            "date", help="End date until which we will download all the missing deltas."
```

Comment:
Having a type check here would help a lot with both using and using and reading the code.

Reply:
This is done.